src/main/ApplicationImpl.cpp

```diff
@@ -333,6 +360,15 @@ ApplicationImpl::~ApplicationImpl()
     {
         mProcessManager->shutdown();
     }
+    if (!modeHasDatabase(mAppMode))
```

Comment:
Because at least in _this_ PR we're treating "no-database" mode as meaning "form buckets while you're running to check hashes, but don't keep buckets around when you shut down". This is debatable.